datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -180,6 +188,81 @@ mod tests {
         Ok((csv, schema))
     }
 
+    #[tokio::test]
+    async fn window_function_with_udaf() -> Result<()> {
+        #[derive(Debug)]
+        struct MyCount(i64);
+
+        impl Accumulator for MyCount {
+            fn state(&self) -> Result<Vec<AggregateState>> {
+                Ok(vec![AggregateState::Scalar(ScalarValue::Int64(Some(
+                    self.0,
+                )))])
+            }
+
+            fn update_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
+                let array = &values[0];
+                self.0 += (array.len() - array.data().null_count()) as i64;
+                Ok(())
+            }
+
+            fn merge_batch(&mut self, states: &[ArrayRef]) -> Result<()> {
+                let counts = downcast_value!(states[0], Int64Array);
```

Comment:
You can use `as_int64_array` for better downcasting schema as it is discussed in #3152